test/e2e/src/core/e2e_tests.sh

```diff
@@ -51,18 +51,67 @@ waitForArcK8sClusterCreated() {
     echo "Arc K8s cluster connectivityState: $connectivityState"
 }
 
+waitForLCMCreated() {
+    connectivityState=false
+    max_retries=60
+    sleep_seconds=10
+    for i in $(seq 1 $max_retries)
+    do
+      echo "iteration: ${i}, clustername: ${CLUSTER_NAME}, resourcegroup: ${RESOURCE_GROUP}"
+      ## hybridaks does not support checking connectivityStatus  maybe compare date with system date to see if has difference
+      # az hybridaks show -n $k8sClusterName -g $resourceGroup --query properties.status.featuresStatus.arcAgentStatus.lastConnectivityTime -o json
+
+      clusterState=$(az connectedk8s show --name $CLUSTER_NAME --resource-group $RESOURCE_GROUP --query connectivityStatus -o json)  arc
+      clusterState=$(echo $clusterState | tr -d '"' | tr -d '"\r\n')
+      echo "cluster current state: ${clusterState}"
+      if [ ! -z "$clusterState" ]; then
+         if [[ ("${clusterState}" == "Connected") || ("${clusterState}" == "Connecting") ]]; then
+            connectivityState=true
+            break
+         fi
+      fi
+      sleep ${sleep_seconds}
+    done
+    echo "Provisioned cluster connectivityState: $connectivityState"
+}
+
 waitForCIExtensionInstalled() {
     installedState=false
     max_retries=60
     sleep_seconds=10
     for i in $(seq 1 $max_retries)
     do
       echo "iteration: ${i}, clustername: ${CLUSTER_NAME}, resourcegroup: ${RESOURCE_GROUP}"
-      installState=$(az k8s-extension show  --cluster-name $CLUSTER_NAME --resource-group $RESOURCE_GROUP  --cluster-type connectedClusters --name azuremonitor-containers --query installState -o json)
+      installState=$(az k8s-extension show --cluster-name $CLUSTER_NAME --resource-group $RESOURCE_GROUP  --cluster-type connectedClusters --name azuremonitor-containers --query provisioningState -o json)
       installState=$(echo $installState | tr -d '"' | tr -d '"\r\n')
```

Comment:
why this has been changed from installState to provisioningState for connectedClusters?

Reply:
looks like the k8s-extension team has removed the installState in the parameters, and I also checked that in the portal/ui side, it is using the provisioningState to check the extension status